elasticsearch_dsl/search.py

```diff
@@ -616,9 +616,23 @@ def suggest(self, name, text, **kwargs):
 
             s = Search()
             s = s.suggest('suggestion-1', 'Elasticsearch', term={'field': 'body'})
+
         """
+        if text is None and regex is None:
+            raise ValueError('You have to pass "text" or "regex" argument.')
+
         s = self._clone()
-        s._suggest[name] = {'text': text}
+
+        query_name = 'text'
+        query_text = text
+        if 'completion' in kwargs:
+            if text:
+                query_name= 'prefix'
+            elif regex:
+                query_name= 'regex'
+                query_text = regex
+
+        s._suggest[name] = {query_name: query_text}
```

Comment:
once we have validated out all the invalid combinations we should be ok with simpler code:
```python
d = s._suggest[name] = kwargs
if text:
    d['text'] = text
else:
    d['regex'] = regex
```